### ISSUE TITLE
Updating ts-loader from 0.8.2 to 0.9.* breaks webpack hot module replacement

### ISSUE THREAD

**Issue by dopare** [TypeStrong/ts-loader#321]
I am using this [angular2 template](https://github.com/aspnet/JavaScriptServices/tree/dev/templates/Angular2Spa)
When I updated ts-loader to 0.9.* it broke hot module replacement. Now all modules are rebuilding on every change. 

With 0.8.2:

```
[HMR] connected
client.js:128 [HMR] bundle rebuilding
client.js:132 [HMR] bundle rebuilt in 224ms
process-update.js:27 [HMR] Checking for updates on the server...
process-update.js:100 [HMR] Updated modules:
process-update.js:102 [HMR]  - ./ClientApp/app/components/counter/counter.component.html
process-update.js:102 [HMR]  - ./ClientApp/app/components/counter/counter.component.ts
process-update.js:102 [HMR]  - ./ClientApp/app/app.module.ts
process-update.js:102 [HMR]  - ./ClientApp/boot-client.ts
process-update.js:107 [HMR] App is up to date.
```
With 0.9.3:
```
[HMR] connected
client.js:128 [HMR] bundle rebuilding
client.js:132 [HMR] bundle rebuilt in 5264ms
process-update.js:27 [HMR] Checking for updates on the server...
process-update.js:100[HMR] Updated modules:
process-update.js:102[HMR]  - ./ClientApp/boot-client.ts
process-update.js:102[HMR]  - ./ClientApp/app/app.module.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/app/app.component.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/navmenu/navmenu.component.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/home/home.component.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/fetchdata/fetchdata.component.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/counter/counter.component.ts
process-update.js:102[HMR]  - ./ClientApp/app/components/counter/counter.component.html
process-update.js:107[HMR] App is up to date.
```

**Comment by johnnyreilly** [comment]
Are you able to confirm with what version of 0.9 this broke?  0.9.0 / 0.9.1 / 0.9.2 / 0.9.3 ?

**Comment by dopare** [comment]
I have checked 0.9.0 and 0.9.3

**Comment by johnnyreilly** [comment]
Hmmm.  I wonder if it's this:

https://github.com/TypeStrong/ts-loader/pull/167/commits/ae824b2676b226bdd0c860a787754a4ae28e339c

**Comment by dopare** [comment]
I reverted the change and it worked.

So that change is the issue.

**Comment by johnnyreilly** [comment]
which one?

**Comment by dopare** [comment]
[ae824b2](https://github.com/TypeStrong/ts-loader/pull/167/commits/ae824b2676b226bdd0c860a787754a4ae28e339c)

**Comment by johnnyreilly** [comment]
Cool - that was just an optimisation a community member supplied.  I'll plan to back it out in the next release. (or make it opt - in only... not decided).  It would be good to have a test that covered this....

**Comment by johnnyreilly** [comment]
Feel free to submit a PR BTW - will likely be doing a release later this week

**Comment by johnnyreilly** [comment]
Released 0.9.4

**Comment by dopare** [comment]
Great, thanks
